--- boost_1_68_0.orig/boost/iostreams/detail/config/fpos.hpp	2018-08-01 22:50:55.000000000 +0200
+++ boost_1_68_0/boost/iostreams/detail/config/fpos.hpp	2018-09-30 13:48:02.395309975 +0200
@@ -25,7 +25,8 @@
 #include <boost/config.hpp>
 
 # if (defined(_YVALS) || defined(_CPPLIB_VER)) && !defined(__SGI_STL_PORT) && \
-     !defined(_STLPORT_VERSION) && !defined(__QNX__) && !defined(_VX_CPU) && !defined(__VXWORKS__)
+     !defined(_STLPORT_VERSION) && !defined(__QNX__) && !defined(_VX_CPU) && !defined(__VXWORKS__) \
+     && !(defined(BOOST_MSVC) && _MSVC_STL_VERSION >= 141)
      /**/
      
 #include <boost/iostreams/detail/ios.hpp>
