diff -ru cppunit-1.12.1/include/cppunit/extensions/HelperMacros.h CppUnit/include/cppunit/extensions/HelperMacros.h
--- cppunit-1.12.1/include/cppunit/extensions/HelperMacros.h	2004-03-13 13:52:56.000000000 +0100
+++ CppUnit/include/cppunit/extensions/HelperMacros.h	2018-03-02 13:18:25.487754669 +0100
@@ -169,7 +169,7 @@
     static CPPUNIT_NS::TestSuite *suite()                                      \
     {                                                                          \
       const CPPUNIT_NS::TestNamer &namer = getTestNamer__();                   \
-      std::auto_ptr<CPPUNIT_NS::TestSuite> suite(                              \
+      std::unique_ptr<CPPUNIT_NS::TestSuite> suite(                            \
              new CPPUNIT_NS::TestSuite( namer.getFixtureName() ));             \
       CPPUNIT_NS::ConcretTestFixtureFactory<TestFixtureType> factory;          \
       CPPUNIT_NS::TestSuiteBuilderContextBase context( *suite.get(),           \
diff -ru cppunit-1.12.1/src/cppunit/TypeInfoHelper.cpp CppUnit/src/cppunit/TypeInfoHelper.cpp
--- cppunit-1.12.1/src/cppunit/TypeInfoHelper.cpp	2004-02-18 21:45:44.000000000 +0100
+++ CppUnit/src/cppunit/TypeInfoHelper.cpp	2018-03-02 13:16:03.100753385 +0100
@@ -7,6 +7,7 @@
 
 #if CPPUNIT_HAVE_GCC_ABI_DEMANGLE
 #include <cxxabi.h>
+#include <stdlib.h>
 #endif
