diff --git a/CTK-0/Libs/PluginFramework/ctkDefaultApplicationLauncher.cpp b/CTK/Libs/PluginFramework/ctkDefaultApplicationLauncher.cpp
index fe37939..48a6fa1 100644
--- a/Libs/PluginFramework/ctkDefaultApplicationLauncher.cpp
+++ b/Libs/PluginFramework/ctkDefaultApplicationLauncher.cpp
@@ -80,18 +80,7 @@ QVariant ctkDefaultApplicationLauncher::start(const QVariant& defaultContext)
   bool doRelaunch = false;
   do
   {
-    try
-    {
-      result = runApplication(defaultContext);
+    result = runApplication(defaultContext);
-    }
-    catch (const std::exception& e)
-    {
-      if (!relaunch || (context->getPlugin()->getState() != ctkPlugin::ACTIVE))
-      {
-        throw;
-      }
-      qWarning() << "Application error:" << e.what();
-    }
     doRelaunch = (relaunch && context->getPlugin()->getState() == ctkPlugin::ACTIVE) ||
                   ctkPluginFrameworkProperties::getProperty(ctkPluginFrameworkLauncher::PROP_OSGI_RELAUNCH).toBool();
   }
diff --git a/CTK-0/Libs/PluginFramework/ctkPluginFrameworkLauncher.cpp b/CTK/Libs/PluginFramework/ctkPluginFrameworkLauncher.cpp
index 9597088..7992ed1 100644
--- a/Libs/PluginFramework/ctkPluginFrameworkLauncher.cpp
+++ b/Libs/PluginFramework/ctkPluginFrameworkLauncher.cpp
@@ -417,32 +417,12 @@ QVariant ctkPluginFrameworkLauncher::run(QRunnable* endSplashHandler, const QVar
     };
     Finalize finalizer;
     Q_UNUSED(finalizer)
-    try
-    {
-      startup(endSplashHandler);
-      if (ctkPluginFrameworkProperties::getProperty(PROP_IGNOREAPP).toBool() || d->isForcedRestart())
-      {
-        return argument;
-      }
-      return run(argument);
+    startup(endSplashHandler);
+    if (ctkPluginFrameworkProperties::getProperty(PROP_IGNOREAPP).toBool() || d->isForcedRestart())
+    {
+      return argument;
+    }
+    return run(argument);
-    }
-    catch (const std::exception& e)
-    {
-      // ensure the splash screen is down
-      if (endSplashHandler != NULL)
-      {
-        endSplashHandler->run();
-      }
-      // may use startupFailed to understand where the error happened
-      if (const ctkException* ce = dynamic_cast<const ctkException*>(&e))
-      {
-        qWarning() << "Startup error:" << ce->printStackTrace();
-      }
-      else
-      {
-        qWarning() << "Startup error:" << e.what();
-      }
-    }
   }
 
   // we only get here if an error happened
@@ -492,11 +472,6 @@ QVariant ctkPluginFrameworkLauncher::run(const QVariant& argument)
     qWarning() << "Application launch failed:" << e.printStackTrace();
     throw;
   }
-  catch (const std::exception& e)
-  {
-    qWarning() << "Application launch failed:" << e.what();
-    throw;
-  }
 }
 
 //----------------------------------------------------------------------------
