--- a/Libs/PluginFramework/ctkPluginAbstractTracked.tpp	2017-10-18 15:56:16.502895134 +0400
+++ b/Libs/PluginFramework/ctkPluginAbstractTracked.tpp	2017-10-18 15:57:06.119974581 +0400
@@ -231,7 +231,7 @@
     qDebug() << "ctkPluginAbstractTracked::untrack[removed]: " << item;
   }
   /* Call customizer outside of synchronized region */
-  customizerRemoved(item, related, object);
+  //customizerRemoved(item, related, object);
   /*
    * If the customizer throws an unchecked exception, it is safe to let it
    * propagate
