diff --git a/CMake/GenerateDCMTKConfigure.cmake b/CMake/GenerateDCMTKConfigure.cmake
index c936b4d..d82781b 100644
--- a/CMake/GenerateDCMTKConfigure.cmake
+++ b/CMake/GenerateDCMTKConfigure.cmake
@@ -548,6 +548,18 @@ ENDIF(DCMTK_WITH_OPENSSL)
 INCLUDE(${DCMTK_CMAKE_INCLUDE}CMake/dcmtkTryCompile.cmake)
 INCLUDE(${DCMTK_CMAKE_INCLUDE}CMake/dcmtkTryRun.cmake)

+IF(NOT (HAVE_ISNAN OR HAVE_PROTOTYPE_ISNAN OR HAVE_PROTOTYPE_STD__ISNAN))
+  unset(HAVE_PROTOTYPE_STD__ISNAN CACHE)
+  DCMTK_TRY_COMPILE(HAVE_PROTOTYPE_STD__ISNAN "std::isnan is provided as function template"
+    "#include <cmath>
+
+    int main()
+    {
+      std::isnan(0.0);
+      return 0;
+    }")
+ENDIF(NOT (HAVE_ISNAN OR HAVE_PROTOTYPE_ISNAN OR HAVE_PROTOTYPE_STD__ISNAN))
+
 IF(NOT DEFINED C_CHAR_UNSIGNED)
   MESSAGE(STATUS "Checking signedness of char")
   DCMTK_TRY_RUN(C_CHAR_SIGNED C_CHAR_SIGNED_COMPILED ${CMAKE_BINARY_DIR}/CMakeTmp/Char
