diff -urNb eigen-eigen-36fd1ba04c12/CMakeLists.txt ep/src/Eigen/CMakeLists.txt
--- eigen-eigen-36fd1ba04c12/CMakeLists.txt	2014-12-16 18:30:52.000000000 +0100
+++ ep/src/Eigen/CMakeLists.txt	2015-01-18 22:53:01.737870010 +0100
@@ -301,7 +301,7 @@
   )
 else()
   set(INCLUDE_INSTALL_DIR
-    "${CMAKE_INSTALL_PREFIX}/include/eigen3"
+    "include/eigen3"
     CACHE INTERNAL
     "The directory where we install the header files (internal)"
   )
