commit 60bdeceff015258608f9d798268853ec4a76fc7d
Author: Christoph Kolb <c.kolb@dkfz-heidelberg.de>
Date:   Wed May 18 12:13:37 2016 +0200

    Prevent building man pages

diff --git a/Utilities/doxygen/CMakeLists.txt b/Utilities/doxygen/CMakeLists.txt
index cf8f616..65608fd 100644
--- a/Utilities/doxygen/CMakeLists.txt
+++ b/Utilities/doxygen/CMakeLists.txt
@@ -224,79 +224,3 @@ ${CMAKE_CURRENT_BINARY_DIR}/latex/Makefile
     subdirs(vtk)
   endif()
 endif()
-
-set(MANPAGES_XML
-    gdcm2pnm
-    gdcm2vtk
-    gdcmanon
-    gdcmconv
-    gdcmdiff
-    gdcmdump
-    gdcmgendir
-    gdcmimg
-    gdcminfo
-    gdcmpap3
-    gdcmpdf
-    gdcmraw
-    gdcmscanner
-    gdcmscu
-    gdcmtar
-    gdcmviewer
-    gdcmxml
-    )
-find_package(LibXslt)
-# need an XSLT 1.0 processor, use xsltproc, maybe add more implementations later
-if(LIBXSLT_XSLTPROC_EXECUTABLE)
-  set(XSLT_PROCESSOR ${LIBXSLT_XSLTPROC_EXECUTABLE})
-# http://docbook.sourceforge.net/release/xsl/current/doc/manpages/man.output.quietly.html
-  set(XSLT_PROCESSOR_ARG --param man.output.quietly 1)
-  # User can change the behavior at cmake time:
-  if(NOT DEFINED GDCM_MANPAGES_USE_NONET)
-    set(GDCM_MANPAGES_USE_NONET FALSE)
-    # By default I want nonet on Debian:
-    if(EXISTS /etc/xml/catalog)
-      set(GDCM_MANPAGES_USE_NONET TRUE)
-    endif()
-  endif()
-  if(GDCM_MANPAGES_USE_NONET)
-    list(APPEND XSLT_PROCESSOR_ARG --nonet)
-  endif()
-else()
-  find_program(MSXSL_EXECUTABLE
-    msxsl
-    )
-  if(MSXSL_EXECUTABLE)
-    set(XSLT_PROCESSOR ${MSXSL_EXECUTABLE})
-    set(XSLT_PROCESSOR_ARG "-pi")
-# msxsl.exe -pi gdcmimg.xml
-#
-#Error occurred while executing stylesheet 'http://docbook.sourceforge.net/release/xsl-ns/current/manpages/docbook.xsl'.
-#
-#Code:   0x80004005
-#Namespace 'http://exslt.org/common' does not contain any functions.
-# -> http://stackoverflow.com/a/16605034/136285 ??
-  endif()
-endif()
-if(XSLT_PROCESSOR)
-foreach(docbook ${MANPAGES_XML})
-  add_custom_command(
-    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${docbook}.1
-    COMMAND ${XSLT_PROCESSOR} ${XSLT_PROCESSOR_ARG} ${CMAKE_CURRENT_SOURCE_DIR}/man/${docbook}.xml
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/man/${docbook}.xml
-    )
-  list(APPEND MANPAGES
-    ${CMAKE_CURRENT_BINARY_DIR}/${docbook}.1
-    )
-endforeach()
-add_custom_target(DOCBOOK_MANPAGES
-  ALL
-  DEPENDS ${MANPAGES}
-  COMMENT  "docbook manpages"
-  )
-install(FILES
-  ${MANPAGES}
-  DESTINATION ${GDCM_INSTALL_MAN_DIR}/man1 COMPONENT DebugDevel
-)
-else()
-  message(WARNING "Cannot build man page from docbook (need an XSL processor)")
-endif()
