diff -rupN original/Modules/ThirdParty/VNL/src/vxl/vcl/tests/test_preprocessor.cxx patch/Modules/ThirdParty/VNL/src/vxl/vcl/tests/test_preprocessor.cxx
--- original/Modules/ThirdParty/VNL/src/vxl/vcl/tests/test_preprocessor.cxx	2015-05-01 02:28:32.000000000 +0200
+++ patch/Modules/ThirdParty/VNL/src/vxl/vcl/tests/test_preprocessor.cxx	2015-05-04 09:39:09.865493336 +0200
@@ -101,6 +101,15 @@ int test_preprocessor_main(int /*argc*/,
   ++minor_count;
 #endif
 
+#ifdef VCL_GCC_5
+  ++major_count;
+#endif
+#ifdef VCL_GCC_50
+  ++minor_count;
+#endif
+#ifdef VCL_GCC_51
+  ++minor_count;
+#endif
 
 #ifdef VCL_VC
   ++compiler_count;
diff -rupN original/Modules/ThirdParty/VNL/src/vxl/vcl/vcl_compiler.h patch/Modules/ThirdParty/VNL/src/vxl/vcl/vcl_compiler.h
--- original/Modules/ThirdParty/VNL/src/vxl/vcl/vcl_compiler.h	2015-05-01 02:28:32.000000000 +0200
+++ patch/Modules/ThirdParty/VNL/src/vxl/vcl/vcl_compiler.h	2015-05-04 09:42:18.201517519 +0200
@@ -126,6 +126,13 @@
 #  else
 #   define VCL_GCC_40
 #  endif
+# elif (__GNUC__==5)
+#  define VCL_GCC_5
+#  if (__GNUC_MINOR__ > 0 )
+#   define VCL_GCC_51
+#  else
+#   define VCL_GCC_50
+#  endif
 # else
 #  error "Dunno about this gcc"
 # endif
@@ -245,7 +252,7 @@
 
 // This *needs* to come after vcl_config_headers.h
 #if defined(__GNUC__) && !defined(__INTEL_COMPILER)
-# if defined(VCL_GCC_3) || defined(VCL_GCC_4)
+# if defined(VCL_GCC_3) || defined(VCL_GCC_4) || defined(VCL_GCC_5)
 #  define GNU_LIBSTDCXX_V3 1
 # elif !defined(GNU_LIBSTDCXX_V3) && defined(VCL_GCC_295) && VCL_CXX_HAS_HEADER_ISTREAM
 // One difference between v2 and v3 is that the former has
