diff --git a/Modules/Video/BridgeOpenCV/include/itkOpenCVVideoIO.h b/Modules/Video/BridgeOpenCV/include/itkOpenCVVideoIO.h
index 1570a182..96a70f17 100644
--- a/Modules/Video/BridgeOpenCV/include/itkOpenCVVideoIO.h
+++ b/Modules/Video/BridgeOpenCV/include/itkOpenCVVideoIO.h
@@ -168,7 +168,7 @@ public:
                       const std::vector<SizeValueType> & dim,
                       const char *                       fourCC,
                       unsigned int                       nChannels,
-                      IOComponentType                    componentType);
+                      IOComponentEnum                    componentType);
 
 protected:
   OpenCVVideoIO();
diff --git a/Modules/Video/BridgeOpenCV/src/itkOpenCVVideoIO.cxx b/Modules/Video/BridgeOpenCV/src/itkOpenCVVideoIO.cxx
index 7d7db3b8..8ce71c8f 100644
--- a/Modules/Video/BridgeOpenCV/src/itkOpenCVVideoIO.cxx
+++ b/Modules/Video/BridgeOpenCV/src/itkOpenCVVideoIO.cxx
@@ -436,7 +436,7 @@ OpenCVVideoIO::SetWriterParameters(TemporalRatioType                  fps,
                                    const std::vector<SizeValueType> & dim,
                                    const char *                       fourCC,
                                    unsigned int                       nChannels,
-                                   IOComponentType                    componentType)
+                                   IOComponentEnum                    componentType)
 {
   if (this->m_ReaderOpen || this->m_WriterOpen)
   {
@@ -444,7 +444,7 @@ OpenCVVideoIO::SetWriterParameters(TemporalRatioType                  fps,
   }
 
   // Make sure componentType is acceptable (right now we only support char)
-  if (componentType != UCHAR)
+  if (componentType != IOComponentEnum::UCHAR)
   {
     itkExceptionMacro("OpenCV IO only supports writing video with pixels of UCHAR");
   }
