diff --git a/Modules/ThirdParty/GDCM/CMakeLists.txt b/Modules/ThirdParty/GDCM/CMakeLists.txt
index bf2d27d..e321215 100644
--- a/Modules/ThirdParty/GDCM/CMakeLists.txt
+++ b/Modules/ThirdParty/GDCM/CMakeLists.txt
@@ -4,7 +4,7 @@ set(ITKGDCM_THIRD_PARTY 1)
 if(ITK_USE_SYSTEM_GDCM)
   set(ITKGDCM_SYSTEM_INCLUDE_DIRS ${GDCM_INCLUDE_DIRS})
   set(ITKGDCM_SYSTEM_LIBRARY_DIRS ${GDCM_LIBRARY_DIRS})
-  set(ITKGDCM_LIBRARIES gdcmDICT gdcmMSFF)
+  set(ITKGDCM_LIBRARIES gdcmDICT gdcmMSFF gdcmDSED)
   set(ITKGDCM_NO_SRC 1)
 
   # When this module is loaded by an app, load GDCM too.
@@ -42,7 +42,7 @@ else()
       endif()
     endif()
   endif()
-  set(ITKGDCM_LIBRARIES  gdcmDICT gdcmMSFF )
+  set(ITKGDCM_LIBRARIES  gdcmDICT gdcmMSFF gdcmDSED)
 endif()
 
 itk_module_impl()
