diff --git a/Modules/ThirdParty/GDCM/CMakeLists.txt b/Modules/ThirdParty/GDCM/CMakeLists.txt
--- a/Modules/ThirdParty/GDCM/CMakeLists.txt
+++ b/Modules/ThirdParty/GDCM/CMakeLists.txt
@@ -4,7 +4,7 @@ if(ITK_USE_SYSTEM_GDCM)
 if(ITK_USE_SYSTEM_GDCM)
   set(ITKGDCM_SYSTEM_INCLUDE_DIRS ${GDCM_INCLUDE_DIRS})
   set(ITKGDCM_SYSTEM_LIBRARY_DIRS ${GDCM_LIBRARY_DIRS})
-  set(ITKGDCM_LIBRARIES gdcmDICT gdcmMSFF)
+  set(ITKGDCM_LIBRARIES gdcmDICT gdcmMSFF gdcmDSED)
   set(ITKGDCM_NO_SRC 1)

   # When this module is loaded by an app, load GDCM too.
@@ -42,7 +42,7 @@ else()
       endif()
     endif()
   endif()
-  set(ITKGDCM_LIBRARIES  gdcmDICT gdcmMSFF )
+  set(ITKGDCM_LIBRARIES  gdcmDICT gdcmMSFF gdcmDSED)
 endif()

 itk_module_impl()
diff --git a/Modules/ThirdParty/Expat/src/expat/CMakeLists.txt b/Modules/ThirdParty/Expat/src/expat/CMakeLists.txt
--- a/Modules/ThirdParty/Expat/src/expat/CMakeLists.txt
+++ b/Modules/ThirdParty/Expat/src/expat/CMakeLists.txt
@@ -38,5 +38,6 @@ INSTALL(TARGETS ITKEXPAT
 INSTALL(FILES
   ${ITK3P_EXPAT_BINARY_DIR}/expatDllConfig.h
   ${ITK3P_EXPAT_SOURCE_DIR}/expat.h
+  ${ITK3P_EXPAT_SOURCE_DIR}/itk_expat_mangle.h
   DESTINATION ${ITK3P_INSTALL_INCLUDE_DIR} # TODO: itk_expat.h #include "itkexpat/expat.h"
   COMPONENT Development)
