diff -burN ITK-4.5.1/CMake/UseITK.cmake ITK/CMake/UseITK.cmake
--- ITK-4.5.1/CMake/UseITK.cmake	2014-02-28 19:46:45.000000000 +0100
+++ ITK/CMake/UseITK.cmake	2015-01-29 14:31:00.137911666 +0100
@@ -100,7 +100,11 @@
   unset(LIST_OF_FACTORY_NAMES)
 
   #-------------------
+  if(NOT NO_DIRECTORY_SCOPED_ITK_COMPILE_DEFINITION)
+    # We add a target scoped compile definition in MITK manually, to avoid leaking
+    # the definition to other targets in the same directory or sub-directory.
   set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS ITK_IO_FACTORY_REGISTER_MANAGER)
+  endif()
   include_directories(BEFORE ${CMAKE_CURRENT_BINARY_DIR}/ITKIOFactoryRegistration)
 
 endif()
