diff -burN OpenIGTLink-master/OpenIGTLinkConfig.cmake.in OpenIGTLink/OpenIGTLinkConfig.cmake.in
--- OpenIGTLink-master/OpenIGTLinkConfig.cmake.in	2014-08-21 18:53:57.000000000 +0200
+++ OpenIGTLink/OpenIGTLinkConfig.cmake.in	2015-02-05 01:08:29.885874473 +0100
@@ -48,7 +48,7 @@
 # A list of all libraries for OpenIGTLink.  Those listed here should
 # automatically pull in their dependencies.
 #SET(OpenIGTLink_LIBRARIES OpenIGTLinkAlgorithms OpenIGTLinkStatistics OpenIGTLinkFEM)
-SET(OpenIGTLink_LIBRARIES OpenIGTLink)
+SET(OpenIGTLink_LIBRARIES optimized;OpenIGTLink;debug;OpenIGTLink@CMAKE_DEBUG_POSTFIX@)
 
 # The OpenIGTLink library dependencies.
 IF(NOT OpenIGTLink_NO_LIBRARY_DEPENDS AND
diff -burN OpenIGTLink-master/Source/CMakeLists.txt OpenIGTLink/Source/CMakeLists.txt
--- OpenIGTLink-master/Source/CMakeLists.txt	2014-08-21 18:53:57.000000000 +0200
+++ OpenIGTLink/Source/CMakeLists.txt	2015-02-05 00:34:34.086991641 +0100
@@ -65,6 +65,7 @@
   igtlutil/igtl_image.h
   igtlutil/igtl_position.h
   igtlutil/igtl_transform.h
+  igtlutil/igtl_status.h
   igtlutil/igtl_types.h
   igtlutil/igtl_util.h
   igtlutil/igtl_capability.h

diff -burN OpenIGTLink-master/Source/igtlSocket.cxx OpenIGTLink/Source/igtlSocket.cxx
--- OpenIGTLink-master/Source/igtlSocket.cxx	2014-08-21 18:53:57.000000000 +0200
+++ OpenIGTLink/Source/igtlSocket.cxx	2015-02-05 00:34:34.086991641 +0100
@@ -51,7 +51,7 @@
 #define WSA_VERSION MAKEWORD(1,1)
 #define igtlCloseSocketMacro(sock) (closesocket(sock))
 #else
-#define igtlCloseSocketMacro(sock) (shutdown(sock, 2))
+#define igtlCloseSocketMacro(sock) ({shutdown(sock, 2); close(sock);})
 #endif

 namespace igtl
