diff --git a/Poco0/Util/src/Application.cpp b/Poco/Util/src/Application.cpp
index 6f60ff9..107ec11 100644
--- a/Util/src/Application.cpp
+++ b/Util/src/Application.cpp
@@ -336,14 +336,6 @@ int Application::run()
 	{
 		logger().log(exc);
 	}
-	catch (std::exception& exc)
-	{
-		logger().error(exc.what());
-	}
-	catch (...)
-	{
-		logger().fatal("system exception");
-	}
 
 	uninitialize();
 	return rc;
