diff -urNb poco-1.6.0-all/cmake/PocoConfig.cmake.in Poco/cmake/PocoConfig.cmake.in
--- poco-1.6.0-all/cmake/PocoConfig.cmake.in	2014-12-22 09:04:42.000000000 +0100
+++ Poco/cmake/PocoConfig.cmake.in	2015-02-04 04:12:45.464421452 +0100
@@ -21,6 +21,10 @@
 
 set(_Poco_NOTFOUND_MESSAGE)
 
+# Let components find each other, but don't overwrite CMAKE_PREFIX_PATH
+set(_Poco_CMAKE_PREFIX_PATH_old ${CMAKE_PREFIX_PATH})
+set(CMAKE_PREFIX_PATH ${_Poco_install_prefix})
+
 foreach(module ${Poco_FIND_COMPONENTS})
     find_package(Poco${module}
         ${_Poco_FIND_PARTS_QUIET}
@@ -39,6 +43,9 @@
     list(APPEND Poco_LIBRARIES "Poco::${module}")
 endforeach()
 
+# Restore the original CMAKE_PREFIX_PATH value
+set(CMAKE_PREFIX_PATH ${_Poco_CMAKE_PREFIX_PATH_old})
+
 if (_Poco_NOTFOUND_MESSAGE)
     set(Poco_NOT_FOUND_MESSAGE "${_Poco_NOTFOUND_MESSAGE}")
     set(Poco_FOUND False)
diff -urNb poco-1.6.0-all/CMakeLists.txt Poco/CMakeLists.txt
--- poco-1.6.0-all/CMakeLists.txt	2014-12-22 09:04:42.000000000 +0100
+++ Poco/CMakeLists.txt	2015-02-04 04:11:52.436420457 +0100
@@ -15,9 +15,9 @@
 # Read the version information from the VERSION file
 file (STRINGS "${CMAKE_SOURCE_DIR}/VERSION" PACKAGE_VERSION )
 message(STATUS "Poco package version: ${PACKAGE_VERSION}")
-string(REGEX REPLACE "([0-9]+)\\.[0-9]+\\.[0-9]+" "\\1" CPACK_PACKAGE_VERSION_MAJOR ${PACKAGE_VERSION})
-string(REGEX REPLACE "[0-9]+\\.([0-9])+\\.[0-9]+" "\\1" CPACK_PACKAGE_VERSION_MINOR ${PACKAGE_VERSION})
-string(REGEX REPLACE "[0-9]+\\.[0-9]+\\.([0-9]+)" "\\1" CPACK_PACKAGE_VERSION_PATCH ${PACKAGE_VERSION})
+string(REGEX REPLACE "([0-9]+)\\.[0-9]+\\.[0-9]+.*" "\\1" CPACK_PACKAGE_VERSION_MAJOR ${PACKAGE_VERSION})
+string(REGEX REPLACE "[0-9]+\\.([0-9])+\\.[0-9]+.*" "\\1" CPACK_PACKAGE_VERSION_MINOR ${PACKAGE_VERSION})
+string(REGEX REPLACE "[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" CPACK_PACKAGE_VERSION_PATCH ${PACKAGE_VERSION})
 
 set(COMPLETE_VERSION ${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH})
 set(RELEASE_NAME "Unstable-trunk")
diff -urNb poco-1.6.0-all/Crypto/cmake/PocoCryptoConfig.cmake Poco/Crypto/cmake/PocoCryptoConfig.cmake
--- poco-1.6.0-all/Crypto/cmake/PocoCryptoConfig.cmake	2014-12-22 09:04:51.000000000 +0100
+++ Poco/Crypto/cmake/PocoCryptoConfig.cmake	2015-02-04 04:12:32.200421203 +0100
@@ -1,4 +1,3 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoCryptoTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/Data/cmake/PocoDataConfig.cmake Poco/Data/cmake/PocoDataConfig.cmake
--- poco-1.6.0-all/Data/cmake/PocoDataConfig.cmake	2014-12-22 09:04:52.000000000 +0100
+++ Poco/Data/cmake/PocoDataConfig.cmake	2015-02-04 04:12:32.208421203 +0100
@@ -1,4 +1,3 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoDataTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/Data/MySQL/cmake/PocoDataMySQLConfig.cmake Poco/Data/MySQL/cmake/PocoDataMySQLConfig.cmake
--- poco-1.6.0-all/Data/MySQL/cmake/PocoDataMySQLConfig.cmake	2014-12-22 09:04:55.000000000 +0100
+++ Poco/Data/MySQL/cmake/PocoDataMySQLConfig.cmake	2015-02-04 04:12:32.208421203 +0100
@@ -1,5 +1,4 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 find_dependency(PocoData)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoDataMySQLTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/Data/ODBC/cmake/PocoDataODBCConfig.cmake Poco/Data/ODBC/cmake/PocoDataODBCConfig.cmake
--- poco-1.6.0-all/Data/ODBC/cmake/PocoDataODBCConfig.cmake	2014-12-22 09:04:54.000000000 +0100
+++ Poco/Data/ODBC/cmake/PocoDataODBCConfig.cmake	2015-02-04 04:12:32.208421203 +0100
@@ -1,5 +1,4 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 find_dependency(PocoData)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoDataODBCTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/Data/SQLite/cmake/PocoDataSQLiteConfig.cmake Poco/Data/SQLite/cmake/PocoDataSQLiteConfig.cmake
--- poco-1.6.0-all/Data/SQLite/cmake/PocoDataSQLiteConfig.cmake	2014-12-22 09:04:54.000000000 +0100
+++ Poco/Data/SQLite/cmake/PocoDataSQLiteConfig.cmake	2015-02-04 04:12:32.208421203 +0100
@@ -1,5 +1,4 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 find_dependency(PocoData)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoDataSQLiteTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/JSON/cmake/PocoJSONConfig.cmake Poco/JSON/cmake/PocoJSONConfig.cmake
--- poco-1.6.0-all/JSON/cmake/PocoJSONConfig.cmake	2014-12-22 09:04:48.000000000 +0100
+++ Poco/JSON/cmake/PocoJSONConfig.cmake	2015-02-04 04:12:32.208421203 +0100
@@ -1,4 +1,3 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoJSONTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/MongoDB/cmake/PocoMongoDBConfig.cmake Poco/MongoDB/cmake/PocoMongoDBConfig.cmake
--- poco-1.6.0-all/MongoDB/cmake/PocoMongoDBConfig.cmake	2014-12-22 09:04:55.000000000 +0100
+++ Poco/MongoDB/cmake/PocoMongoDBConfig.cmake	2015-02-04 04:12:32.208421203 +0100
@@ -1,5 +1,4 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 find_dependency(PocoNet)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoMongoDBTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/Net/cmake/PocoNetConfig.cmake Poco/Net/cmake/PocoNetConfig.cmake
--- poco-1.6.0-all/Net/cmake/PocoNetConfig.cmake	2014-12-22 09:04:50.000000000 +0100
+++ Poco/Net/cmake/PocoNetConfig.cmake	2015-02-04 04:12:32.208421203 +0100
@@ -1,4 +1,3 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoNetTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/NetSSL_OpenSSL/cmake/PocoNetSSLConfig.cmake Poco/NetSSL_OpenSSL/cmake/PocoNetSSLConfig.cmake
--- poco-1.6.0-all/NetSSL_OpenSSL/cmake/PocoNetSSLConfig.cmake	2014-12-22 09:04:52.000000000 +0100
+++ Poco/NetSSL_OpenSSL/cmake/PocoNetSSLConfig.cmake	2015-02-04 04:12:32.208421203 +0100
@@ -1,5 +1,4 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 find_dependency(PocoUtil)
 find_dependency(PocoNet)
diff -urNb poco-1.6.0-all/Util/cmake/PocoUtilConfig.cmake Poco/Util/cmake/PocoUtilConfig.cmake
--- poco-1.6.0-all/Util/cmake/PocoUtilConfig.cmake	2014-12-22 09:04:49.000000000 +0100
+++ Poco/Util/cmake/PocoUtilConfig.cmake	2015-02-04 04:12:45.464421452 +0100
@@ -1,5 +1,4 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 find_dependency(PocoXML)
 find_dependency(PocoJSON)
diff -urNb poco-1.6.0-all/XML/cmake/PocoXMLConfig.cmake Poco/XML/cmake/PocoXMLConfig.cmake
--- poco-1.6.0-all/XML/cmake/PocoXMLConfig.cmake	2014-12-22 09:04:47.000000000 +0100
+++ Poco/XML/cmake/PocoXMLConfig.cmake	2015-02-04 04:12:45.464421452 +0100
@@ -1,4 +1,3 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 include("${CMAKE_CURRENT_LIST_DIR}/PocoXMLTargets.cmake")
\ No newline at end of file
diff -urNb poco-1.6.0-all/Zip/cmake/PocoZipConfig.cmake Poco/Zip/cmake/PocoZipConfig.cmake
--- poco-1.6.0-all/Zip/cmake/PocoZipConfig.cmake	2014-12-22 09:04:55.000000000 +0100
+++ Poco/Zip/cmake/PocoZipConfig.cmake	2015-02-04 04:12:45.464421452 +0100
@@ -1,5 +1,4 @@
 include(CMakeFindDependencyMacro)
-set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_LIST_DIR})
 find_dependency(PocoFoundation)
 find_dependency(PocoUtil)
 find_dependency(PocoXML)
