diff -urNb SimpleITK-0.8.0/CMakeLists.txt SimpleITK-src/CMakeLists.txt
--- SimpleITK-0.8.0/CMakeLists.txt	2014-03-14 13:36:01.000000000 +0100
+++ SimpleITK-src/CMakeLists.txt	2015-01-11 15:25:33.237871581 +0100
@@ -51,6 +51,7 @@
 
 find_package(ITK REQUIRED ) # the modules needed can be listed here as required components
 if(ITK_FOUND)
+  find_package(GDCM PATHS ${GDCM_DIR} REQUIRED)
 
   # NOTE: We are purposely not calling UseITK yet. However, we must make
   # sure the requred compilation and linker flags are set. Since, we
@@ -65,7 +66,8 @@
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${ITK_REQUIRED_LINK_FLAGS}")
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${ITK_REQUIRED_LINK_FLAGS}")
 
-  link_directories( "${ITK_LIBRARY_DIRS}")
+  link_directories( "${ITK_LIBRARY_DIRS}" "${GDCM_LIBRARY_DIRS}")
+  list(REMOVE_ITEM ITK_LIBRARIES ITKVideoBridgeOpenCV ITKVideoCore ITKVideoIO)
 
 
 endif()
@@ -514,7 +516,7 @@
 #------------------------------------------------------------------------------
 # CPack
 
-set(CPACK_SOURCE_IGNORE_FILES  "${ITK_MODULES_DISABLED_CPACK};/\\\\.git")
+set(CPACK_SOURCE_IGNORE_FILES  "${ITK_MODULES_DISABLED_CPACK}/\\\\.git")
 set(CPACK_PACKAGE_VERSION_MAJOR "${SimpleITK_Major}")
 set(CPACK_PACKAGE_VERSION_MINOR "${SimpleITK_Minor}")
 set(CPACK_PACKAGE_VERSION_PATCH "${SimpleITK_Patch}")
diff -urNb SimpleITK-0.8.0/SimpleITKConfig.cmake.in SimpleITK-src/SimpleITKConfig.cmake.in
--- SimpleITK-0.8.0/SimpleITKConfig.cmake.in	2014-03-14 13:36:01.000000000 +0100
+++ SimpleITK-src/SimpleITKConfig.cmake.in	2015-01-11 15:25:33.249871581 +0100
@@ -37,14 +37,14 @@
 endif()
 
 # Import ITK targets.
-if(NOT ITK_TARGETS_IMPORTED)
+if(NOT TARGET ITKCommon)
   set(ITK_TARGETS_IMPORTED 1)
   include("${ITK_CONFIG_TARGETS_FILE}")
 endif()
 
 # Import SimpleITK targets.
 set(SimpleITK_TARGETS_FILE "${_SimpleITKConfig_DIR}/SimpleITKTargets.cmake")
-if(NOT SimpleITK_TARGETS_IMPORTED)
+if(NOT TARGET SimpleITKCommon)
   set(SimpleITK_TARGETS_IMPORTED 1)
   include("${SimpleITK_TARGETS_FILE}")
 endif()
