diff --git a/CMakeLists.txt b/CMakeLists.txt
index bae8e43..a5f3070 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,9 +48,11 @@ message(STATUS "Building SimpleITK version \"${SimpleITK_VERSION}\"")
 # done.
 include(sitkCheckRequiredFlags)
 
+FIND_PACKAGE(HDF5)
 
 find_package(ITK REQUIRED ) # the modules needed can be listed here as required components
 if(ITK_FOUND)
+  find_package(GDCM PATHS ${GDCM_DIR} REQUIRED)
 
   # NOTE: We are purposely not calling UseITK yet. However, we must make
   # sure the requred compilation and linker flags are set. Since, we
@@ -65,7 +67,8 @@ if(ITK_FOUND)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${ITK_REQUIRED_LINK_FLAGS}")
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${ITK_REQUIRED_LINK_FLAGS}")
 
-  link_directories( "${ITK_LIBRARY_DIRS}")
+  link_directories( "${ITK_LIBRARY_DIRS}" "${GDCM_LIBRARY_DIRS}")
+  list(REMOVE_ITEM ITK_LIBRARIES ITKVideoBridgeOpenCV ITKVideoCore ITKVideoIO)
 
 
 endif()
@@ -514,7 +517,7 @@ install(FILES ${SimpleITK_DOC_FILES} DESTINATION "${SimpleITK_INSTALL_DOC_DIR}"
 #------------------------------------------------------------------------------
 # CPack
 
-set(CPACK_SOURCE_IGNORE_FILES  "${ITK_MODULES_DISABLED_CPACK};/\\\\.git")
+set(CPACK_SOURCE_IGNORE_FILES  "${ITK_MODULES_DISABLED_CPACK}/\\\\.git")
 set(CPACK_PACKAGE_VERSION_MAJOR "${SimpleITK_Major}")
 set(CPACK_PACKAGE_VERSION_MINOR "${SimpleITK_Minor}")
 set(CPACK_PACKAGE_VERSION_PATCH "${SimpleITK_Patch}")
diff --git a/SimpleITKConfig.cmake.in b/SimpleITKConfig.cmake.in
index f10c2a0..34e076b 100644
--- a/SimpleITKConfig.cmake.in
+++ b/SimpleITKConfig.cmake.in
@@ -37,14 +37,14 @@ if(NOT ITK_CONFIG_TARGETS_FILE)
 endif()
 
 # Import ITK targets.
-if(NOT ITK_TARGETS_IMPORTED)
+if(NOT TARGET ITKCommon)
   set(ITK_TARGETS_IMPORTED 1)
   include("${ITK_CONFIG_TARGETS_FILE}")
 endif()
 
 # Import SimpleITK targets.
 set(SimpleITK_TARGETS_FILE "${_SimpleITKConfig_DIR}/SimpleITKTargets.cmake")
-if(NOT SimpleITK_TARGETS_IMPORTED)
+if(NOT TARGET SimpleITKCommon)
   set(SimpleITK_TARGETS_IMPORTED 1)
   include("${SimpleITK_TARGETS_FILE}")
 endif()
