--- CMakeOptions.cmake	Wed Aug 12 11:43:46 2015
+++ CMakeOptions.cmake	Thu Aug 20 14:25:55 2015
@@ -16,6 +16,15 @@
 ENDIF ( NOT VTK_FOUND )
 
 #
+# Try to find GDCM and include its settings (otherwise complain)
+#
+IF ( NOT GDCM_FOUND )
+  FIND_PACKAGE(GDCM REQUIRED)
+  INCLUDE(${GDCM_USE_FILE})
+ENDIF ( NOT GDCM_FOUND )
+
+
+#
 # Try to find ITK and include its settings (otherwise complain)
 #
 IF ( NOT ITK_FOUND )
